broker-daemon/block-order-worker/index.js

```diff
@@ -570,13 +585,39 @@ class BlockOrderWorker extends EventEmitter {
       }
     })
 
-    // reject the entire block order if an underlying order fails
+    // try to re-place the block order if the relayer drops connection, while allowing the broker to
+    // update block order status (i.e. broker can still cancel orders if relayer goes down)
     osm.once('reject', async () => {
+      if (osm.shouldRetry()) {
+        const retryBlockOrder = async () => {
+          if (!await this.relayerIsAvailable()) {
+            throw new Error('Relayer not available')
+          }
+
+          this.logger.info('Retrying order for block order', { order: osm.order })
+
+          // Get the existing order in case an update to order status happened between retries
+          const updatedBlockOrder = await this.getBlockOrder(blockOrder.id)
```

Comment:
hmm, what's going on here? Shouldn't the block order be passed by reference throughout the worker? What kind of update might happen that would cause us to pull from the data store?